src/model/SplineSegment.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Xournal++
+ *
+ * A spline segment
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <gtk/gtk.h>
+
+#include "model/Element.h"
+#include "model/Stroke.h"
+
+#include "Point.h"
+
+
+/**
+ * @brief A class to handle splines segments
+ *
+ * Every spline segment is defined by two knot point and two control points.
+ * This contrasts to the cairo-draw command cairo_curve_to, where everything is relative to the current position of
+ * another knot,
+ */
+
+
+class SplineSegment {
+public:
+    SplineSegment() = default;
+    ~SplineSegment() = default;
```

Comment:
It doesn't really matter, but I prefer to remove the "custom defaulted destructor" as it makes no difference for the compiler, but many people not using our .clang-tidy file will get into a warning hell, because the "rule of 5" is not applied.